src/core/thread/mesh_forwarder.cpp

```diff
@@ -1056,11 +1056,18 @@ ThreadError MeshForwarder::HandleFrameRequest(Mac::Frame &aFrame)
             }
         }
 
-        if (SendFragment(*mSendMessage, aFrame) == kThreadError_NotCapable)
+        error = SendFragment(*mSendMessage, aFrame);
+
+        // `SendFragment()` fails with `NotCapable` error if the message is MLE (with
+        // no link layer security) and also requires fragmentation.
+        if (error == kThreadError_NotCapable)
         {
-            SendFragment(*mSendMessage, aFrame);
+            // Enable security and try again.
+            mSendMessage->SetLinkSecurityEnabled(true);
+            error = SendFragment(*mSendMessage, aFrame);
         }
 
+        SuccessOrExit(error);
```

Comment:
Yes. There is indeed no error case with current implementation. We can make it `assert` to be safer. I will change it.
It also help making it easier to catch if we later have any other failure paths in `SendFragment()` returning an error (I doubt that we will ever need to though).